packages/price-exporter/src/main.js

```diff
@@ -1,3 +1,270 @@
-export default function addSum (a, b) {
-  return a + b
+/* @flow */
+import type {
+  ApiConfigOptions,
+  Configuration,
+  ExporterOptions,
+  LoggerOptions,
+  ProcessedPriceObject,
+  UnprocessedPriceObject,
+} from 'types/price'
+import type {
+  Client,
+  ClientRequest,
+} from 'types/sdk'
+
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import {
+  createAuthMiddlewareForClientCredentialsFlow,
+} from '@commercetools/sdk-middleware-auth'
+import {
+  createUserAgentMiddleware,
+} from '@commercetools/sdk-middleware-user-agent'
+import csv from 'fast-csv'
+import JSONStream from 'JSONStream'
+import Promise from 'bluebird'
+import { flatten } from 'flat'
+import { flattenDeep } from 'lodash'
+import * as utils from './utils'
+import pkg from '../package.json'
+
+export default class PriceExporter {
+  // Set flowtype annotations
+  apiConfig: ApiConfigOptions;
+  client: Client;
+  config: Configuration;
+  logger: LoggerOptions;
+  _cache: Object;
+  _resolveReferences: Function;
+
+  constructor (
+    options: ExporterOptions,
+    logger: LoggerOptions,
+  ) {
+    if (!options.apiConfig)
+      throw new Error('The constructor must be passed an `apiConfig` object')
+    this.apiConfig = options.apiConfig
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    const defaultOptions = {
+      delimiter: ',',
+      exportFormat: 'json',
+      staged: false,
+    }
+
+    this.config = { ...defaultOptions, ...options }
+
+    this.logger = {
+      error: () => {},
+      warn: () => {},
+      info: () => {},
+      verbose: () => {},
+      ...logger,
+    }
+    this._cache = {}
+
+    this._resolveReferences = this._resolveReferences.bind(this)
+  }
+
+  run (outputStream: stream$Writable) {
+    this.logger.info('Starting Export')
+    const request = this._buildRequest('productProjections')
+    this.client.process(request, (data) => {
+      const products = data.body.results
+      this.logger.verbose(`Fetched ${products.length} products`)
+      let processedBatch = []
+      return Promise.map(products, product => utils._getPrices(product))
+      .then((pricesArray) => {
+        processedBatch = this._handlePrices(pricesArray)
+        return Promise.resolve(processedBatch)
+      })
+    }, { accumulate: true })
+    .then((allPricesBatches) => {
+      this._handleOutput(allPricesBatches, outputStream)
+    })
+    .catch((e) => {
+      outputStream.emit('error', e)
+    })
+  }
+
+  _handlePrices (productPricesArray: Array<UnprocessedPriceObject>) {
```

Comment:
though I understand how hand it can be to come up with method names, this method names are really very vague and does not describe what is being done in the methods.

I think we should strip all _.._handle_. that might help in coming up with better method names.